onnxruntime/python/tools/transformers/optimizer.py

```diff
@@ -290,6 +294,19 @@ def optimize_model(
     """
     assert opt_level is None or opt_level in [0, 1, 2, 99]
 
+    # Auto detect if input model has external data
+    has_external_data_file = False
+    original_model = load_model(input, load_external_data=False)
+    for initializer in original_model.graph.initializer:
+        if initializer.HasField("data_location") and initializer.data_location == TensorProto.EXTERNAL:
+            has_external_data_file = True
+            break
+    del original_model
+
+    if model_type not in MODEL_TYPES:
+        logger.warning(f"Unsupported model type: {model_type} for optimization, directly return model.")
+        return OnnxModel(load_model(input, load_external_data=has_external_data_file))
```

Comment:
Need not check whether the model has external data here. You can always load model with load_external_data=True (even the model does not have external data).

Reply:
Updated.